client/test/registration.test.js

```diff
@@ -0,0 +1,181 @@
+    });
+    it('should not write anything by default', () => {
+      assert.equal(this.reg.saveUUIDSync(), false);
+    });
+
+    describe('with a uuid', () => {
+      beforeEach(() => {
+        this.reg.uuid = 'foobar-uuid';
+      });
+
+      it('should write a uuid.json', () => {
+        assert(this.reg.saveUUIDSync());
+        try {
+          fs.statSync(this.reg.uuidPath());
+        }
+        catch (err) {
+          if (err.code == 'ENOENT') {
+            assert.fail('Could not find the uuid saved on disk');
+          }
+          else {
+            throw err;
+          }
+        }
+      });
+    });
+  });
+
+  describe('saveKeysSync()', () => {
+    it('should return false if there are not keys', () => {
+      const r = new Registration();
+      assert(!r.saveKeysSync());
+    });
+
+    describe('when keys have been generated', () => {
+      beforeEach(() => {
+        this.reg = new Registration();
+        this.reg.generateKeys();
+      });
+      it('should return true if the public key has been written', () => {
+        assert(this.reg.saveKeysSync());
+        try {
+          fs.statSync(this.reg.publicKeyPath());
+        }
+        catch (err) {
+          assert.fail('The public key was not written properly');
+        }
+
+        try {
+          const privateKeyPath = [this.reg.keyPath(),
+            'evergreen-private-key'].join(path.sep);
+
+          fs.statSync(privateKeyPath);
+        }
+        catch (err) {
+          assert.fail('The private key was not written properly');
+        }
+      });
+    });
+  });
+
+  describe('loadKeysSync()', () => {
+    beforeEach(() => {
+      this.reg = new Registration();
+    });
+
+    it('should return false by default when there are no keys', () => {
+      assert.equal(this.reg.loadKeysSync(), false);
+    });
+
+    it('should return false if keys have already been generated', () => {
+      this.reg.generateKeys();
+      assert.equal(this.reg.loadKeysSync(), false);
+    });
+
+    describe('when keys are already on disk', () => {
+      beforeEach(() => {
+        const preflight = new Registration();
+        preflight.generateKeys();
+        preflight.saveKeysSync();
+      });
+
+      it('should return true and have keys loaded if they are on disk', () => {
+        assert(this.reg.loadKeysSync());
+        assert.equal(typeof this.reg.getPublicKey(), 'string');
+      });
+    });
+  });
+
+  describe('hasKeys()', () => {
+    it('should return false by default', () => {
+      assert.equal((new Registration()).hasKeys(), false);
+    });
+  });
+
+  describe('generateKeys()', () => {
+    it('should return a boolean on success', () => {
+      assert.ok((new Registration()).generateKeys());
+    });
+  });
+
+  describe('publicKeyPath()', () => {
+    it('should return a path', () => {
+      const p = (new Registration()).publicKeyPath();
+      assert(p != path.basename(p), 'This doesn\'t look like a path');
```

Comment:
IIUC, the fact you cannot use `"This doesn't look like a path"` here is because of the esling setting done above at https://github.com/jenkins-infra/evergreen/pull/37/files#diff-d89f635f668d643667d79f8fcc5f9f41R20

Reply:
Yeah, and that is okay :smile: